### ISSUE TITLE
Unreasonable use of JSON_FORCE_OBJECT in PersistenceStrategies' implementations

### ISSUE THREAD

**Issue by afoeder** [prooph/pdo-event-store#87]
In every implementation of `\Prooph\EventStore\Pdo\PersistenceStrategy::prepareData` there are json_encode occurrences with the JSON_FORCE_OBJECT flag set:

```
public function prepareData(Iterator $streamEvents): array
    {
        $data = [];

        foreach ($streamEvents as $event) {
            $data[] = $event->uuid()->toString();
            $data[] = $event->messageName();
            $data[] = json_encode($event->payload(), \JSON_FORCE_OBJECT);
            $data[] = json_encode($event->metadata(), \JSON_FORCE_OBJECT);
            $data[] = $event->createdAt()->format('Y-m-d\TH:i:s.u');
        }

        return $data;
    }
```

This results in plain arrays being handled like objects, i.e. `array('hoo', 'ray')` results in `{"0":"hoo","1":"ray"}"` ([3v4l example](https://3v4l.org/ILE47))

From the [PHP docs](http://php.net/manual/en/json.constants.php):

> **JSON_FORCE_OBJECT** Outputs an object rather than an array when a non-associative array is used. Especially useful when the recipient of the output is expecting an object and the array is empty. Available since PHP 5.3.0.

Please reconsider its necessity.

// additional note: decoding it back with second argument (assoc) true will restore a properly indexed array: https://3v4l.org/Brr7i

**Comment by bweston92** [comment]
@afoeder unfortunately PHPUnit thinks it is fine because the keys match. See #91

**Comment by afoeder** [comment]
I disagree with closing it. Just because no presently existing unit test fails or changes doesn't mean the implementation ist correct.
It _coincidentally_ works when reconstituting from the database, but the intermediate format is incorrect as a matter of fact and I propose fixing it...

**Comment by prolic** [comment]
It's already fixed with v1.3.1 incl tests.

**Comment by afoeder** [comment]
sorry @prolic, read that via mobile and didn't see it was _closed through #93_. Thank you!
